Automattic/wp-calypso
# Sharing: Implement Redux-based Publicize reducers and actions

Closes #350
Partially implements #129

This pull request seeks to implement a new set of Redux-based actions, reducers, selectors, and components related to displaying active Publicize connections in the post editor Sharing accordion. This will serve to eliminate an unnecessary request to the REST API for retrieving Keyring connections, as this data is not currently in use by the post editor.

__Implementation notes:__

This is a large pull request, but it is mostly comprised of new modules, of which include significant non-front-facing changes (i.e. documentation, testing).

As this pull request introduces Redux, it also includes the necessary changes to wire a new instance of a global Redux store to the routing context. See `shared/lib/redux-store` . The intention is that as reducers are added, they should be included in the `combineReducers` call in `shared/lib/redux-store/reducers.js` . Similarly, as new actions are introduced, they would be added to `shared/lib/redux-store/action-types.js` . The proposed advantage here is that, by having a single set of action type keys, there's a reduced chance for key conflicts and that this list will serve as a single reference for all action types in use in the application.

[react-redux](https://github.com/rackt/react-redux) is used here for passing the store context through the hierarchy of post editor components. This is used by a new [`<EditorSharingContainer />`](https://github.com/Automattic/wp-calypso/tree/update/sharing-redux/client/post-editor/editor-sharing/container.jsx) component for managing and fetching site connections.

A nested subdirectory was created for Publicize-related files, as it's intended that we'll soon introduce new reducers and actions for managing Keyring connections.

__Testing instructions:__

The changes herein replace the current post editor Sharing accordion connections listing. Verify that existing behavior remains unaffected.

1. Navigate to the [Calypso post editor](http://calypso.localhost:3000/post)
2. Select a site, if prompted
3. Expand the sidebar Sharing accordion
4. Note that connections available to the current user (shared or owned) are listed
5. Click Connect New Service
6. Connect or disconnect to a Publicize service
7. Close the popup window after having completed step 6
8. Note that connections are refreshed and list connections available to the current user
